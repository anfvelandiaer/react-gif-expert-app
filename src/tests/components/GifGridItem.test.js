import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas en <GifGridItem />", () => {
  const title = "Un Titulo";
  const url = "httpos://localhost.com";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("should mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe tener un pÃ¡rrafo con el titulo", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("Debe tener la imagen igual al url y alt de los props", () => {
    const img = wrapper.find("img");
    //Obtener las props html como src, alt ...
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('should tener animate_fadeIn', () => { 
      const div = wrapper.find('div');
      const className = div.prop('className');
     expect(className.includes('animate__fadeIn')).toBe(true)
   })
});
