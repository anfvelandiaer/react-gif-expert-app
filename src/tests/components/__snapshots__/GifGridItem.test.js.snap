// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> should mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un Titulo"
    src="httpos://localhost.com"
  />
  <p>
     
    Un Titulo
     
  </p>
</div>
`;
